cfme/infrastructure/pxe.py

```diff
@@ -24,74 +21,194 @@
-        ('access_url_text', Input('access_url')),
-        ('uri_text', Input('uri')),
-        ('pxe_dir_text', Input('pxe_directory')),
-        ('windows_dir_text', Input('windows_images_directory')),
-        ('customize_dir_text', Input('customization_directory')),
-        ('pxe_menu_text', Input('pxemenu_0')),
-    ])
+        def get_all(self):
+            return [self.entity_type(parent=self, additional_context={'name': name})
+                    for name in self._entity_names()]
 
-pxe_image_type_form = Form(
-    fields=[
-        ('image_type', Select("//select[@id='image_typ']"))
-    ])
+        def get_entity(self, name):
+            return self.entity_type(parent=self, additional_context={'name': name})
 
+    @property
+    def is_displayed(self):
+        return (super(self, PXEServersView).is_displayed and
+                self.entities.title.text == 'All PXE Servers')
 
-template_details_page = Region(infoblock_type='form')  # infoblock shoudl be type 'detail' #gofigure
 
-template_properties_form = Form(
-    fields=[
-        ('name_text', Input('name')),
-        ('description_text', Input('description')),
-        ('image_type', Select('//select[@id="img_typ"]')),
-        ('script_type', Select('//select[@id="typ"]')),
-        ('script_data', ScriptBox(ta_locator="//textarea[@id='script_data']"))
-    ])
+class PXEDetailsToolBar(PXEToolBar):
+    """
+     represents the toolbar which appears when any pxe entity is clicked
+    """
+    reload = Button(title='Reload current display')
+
+
+class PXEServerDetailsView(PXEMainView):
+    """
+     represents Server Details view
+    """
+    toolbar = View.nested(PXEDetailsToolBar)
 
+    @View.nested
+    class entities(View):  # noqa
+        basic_information = SummaryTable(title="Basic Information")
+        pxe_image_menus = SummaryTable(title='PXE Image Menus')
 
-image_table = Table('//div[@id="records_div"]//table')
+    @property
+    def is_displayed(self):
+        return False
 
-image_properties_form = Form(
-    fields=[
-        ('name_text', Input('name')),
-        ('provision_type', Select('//select[@id="provision_type"]'))
-    ])
 
-iso_details_page = Region(infoblock_type='form')  # infoblock shoudl be type 'detail' #gofigure
+class PXEServerAddView(View):
+    """
+      represents Add New PXE Server view
+    """
+    title = Text('//div[@id="main-content"]//h1')
+    flash = FlashMessages('.//div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+                          'contains(@class, "flash_text_div")]')
+    # common fields
+    name = Input(id='name')
+    depot_type = BootstrapSelect(id='log_protocol')
+    access_url = Input(id='access_url')
+    pxe_dir = Input(id='pxe_directory')
+    windows_images_dir = Input(id='windows_images_directory')
+    customization_dir = Input(id='customization_directory')
+    filename = Input(id='pxemenu_0')
+
+    uri = Input(id='uri')  # both NFS and Samba
+
+    # Samba only
```

Comment:
first implementation used ConditinalView. However, when I implemented it, I realized that both views are almost equal. It just doesn't make sense to use it there and play with complex fill dicts and if later. So, I combined both views into one view